### PR TITLE
integration tests: add  IBM VPC support (SC-1352)

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
integration tests: add IBM VPC support
```

## Additional Context

https://warthogs.atlassian.net/browse/SC-1352
https://github.com/canonical/pycloudlib/pull/225

## Test Steps

```sh
CLOUD_INIT_PLATFORM=ibm tox -re integration-tests -- tests/integration_tests/<some-test>
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: tests/integration_tests/clouds.py
Comment (TheRealFalcon) [inline tests/integration_tests/clouds.py:421]: Let's have this follow the <name>-integration-test convention. I.e., `ibm-integration-test`

**Review by TheRealFalcon — APPROVED** [review]
One comment inline. Also, can we add ibm to the list here:
https://github.com/canonical/cloud-init/blob/main/tests/integration_tests/integration_settings.py#L18

Those are fairly trivial changes, so once you make feel free to merge.

File: tests/integration_tests/clouds.py
Comment (aciba90) [reply tests/integration_tests/clouds.py:421]: I forgot to mention that instance/image names starting with `ibm` are reserved in IBM VPC.

Taking into account that, do you think `cloud-init-integration-test-ibm` is okay?

Comment (TheRealFalcon) [reply tests/integration_tests/clouds.py:421]: Ah, sad. That is fine then. It might be worth double checking there's no length restriction in IBM. The name plus the timeestamp could get long. There is a regex here used to validate instance names:
https://github.com/aciba90/pycloudlib/blob/main/pycloudlib/util.py#L322

Comment (aciba90) [reply tests/integration_tests/clouds.py]: * Renamed the tag to `integration-test-ibm`
* Launched an integration test and passes, thus `validate_tag` accepts a tag with timestamp
* The max length of instance / snapshot names in IBM VPC is 63 (see attached image), thus we do not need to restrict the regex within `validate_tag`.

image

**Comment by TheRealFalcon** [comment]
Your changes LGTM. Since this is already approved, feel free to merge when ready.

**Pull request merged**
